projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  Component,
+  ContentChild,
+  HostBinding,
+  Input,
+  OnInit,
+  Optional,
+  Self,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormControl,
+  FormGroup,
+  FormGroupDirective,
+  NgControl,
+  ValidationErrors,
+  Validators,
+} from '@angular/forms';
+
+import { Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+import { LgErrorStateMatcher } from '../validation/error-state-matcher';
+import { LgDomService } from '../../utils/dom.service';
+import { LgHintComponent } from '../hint';
+import { SortCodeField } from './sort-code.interface';
+import { LgValidationComponent } from '../validation';
+
+let nextUniqueId = 0;
+
+@Component({
+  selector: 'lg-sort-code',
+  templateUrl: './sort-code.component.html',
+  styleUrls: ['./sort-code.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
+  @HostBinding('class.lg-sort-code-field') class = true;
+
+  private uniqueId = nextUniqueId++;
+  sortCodeFormGroup: FormGroup;
+  first: FormControl;
+  second: FormControl;
+  third: FormControl;
+  valueChanges: Subscription;
+  hasError: boolean;
+
+  @Input() value: string;
+  @Input() disabled: false;
+  @Input() firstId = `lg-input-sort-code-first-${this.uniqueId++}`;
```

Comment:
Careful here with `++`, I think what you really want is:
```
@Input() firstId = `lg-input-sort-code-first-${this.uniqueId}`;
@Input() secondId = `lg-input-sort-code-second-${this.uniqueId}`;
@Input() thirdId = `lg-input-sort-code-third-${this.uniqueId}`;
``

Reply:
Ah yes this has come up a few times, I totally forgot 🤦

Reply:
Done ✅